testing/util/lightclient.go

```diff
@@ -28,1049 +37,833 @@ type TestLightClient struct {
+	}
+}
 
-		signedBlock, err = blocks.NewSignedBeaconBlock(block)
-		require.NoError(l.T, err)
-
-		h, err := signedBlock.Header()
-		require.NoError(l.T, err)
-
-		err = state.SetLatestBlockHeader(h.Header)
-		require.NoError(l.T, err)
-		stateRoot, err := state.HashTreeRoot(ctx)
-		require.NoError(l.T, err)
-
-		// get a new signed block so the root is updated with the new state root
-		block.Block.StateRoot = stateRoot[:]
-		signedBlock, err = blocks.NewSignedBeaconBlock(block)
-		require.NoError(l.T, err)
-	} else {
-		block := NewBeaconBlockCapella()
-		block.Block.Slot = slot
-		block.Block.ParentRoot = parentRoot[:]
+// WithNoFinalizedCheckpoint avoids setting a finalized checkpoint for the attested state
+func WithNoFinalizedCheckpoint() LightClientOption {
+	return func(l *TestLightClient) {
+		l.noFinalizedCheckpoint = true
+	}
+}
 
-		var trueBitNum uint64
-		if supermajority {
-			trueBitNum = (params.BeaconConfig().SyncCommitteeSize / 3 * 2) + 1
-		} else {
-			trueBitNum = params.BeaconConfig().MinSyncCommitteeParticipants
-		}
-		for i := uint64(0); i < trueBitNum; i++ {
-			block.Block.Body.SyncAggregate.SyncCommitteeBits.SetBitAt(i, true)
+// WithFinalizedCheckpointInPrevFork creates a finalized checkpoint for the attested state, in the previous fork.
+func WithFinalizedCheckpointInPrevFork() LightClientOption {
+	return func(l *TestLightClient) {
+		if l.version == version.Altair {
+			l.T.Fatalf("Can't set finalized checkpoint in previous fork for Altair")
 		}
+		l.finalizedCheckpointInPrevFork = true
+	}
+}
 
-		signedBlock, err = blocks.NewSignedBeaconBlock(block)
-		require.NoError(l.T, err)
-
-		h, err := signedBlock.Header()
-		require.NoError(l.T, err)
-
-		err = state.SetLatestBlockHeader(h.Header)
-		require.NoError(l.T, err)
-		stateRoot, err := state.HashTreeRoot(ctx)
-		require.NoError(l.T, err)
-
-		// get a new signed block so the root is updated with the new state root
-		block.Block.StateRoot = stateRoot[:]
-		signedBlock, err = blocks.NewSignedBeaconBlock(block)
-		require.NoError(l.T, err)
+// WithSupermajority Specifies whether the sync committee bits have supermajority or not
+func WithSupermajority() LightClientOption {
+	return func(l *TestLightClient) {
+		l.supermajority = true
 	}
+}
 
-	l.State = state
-	l.AttestedState = attestedState
-	l.AttestedBlock = signedParent
-	l.Block = signedBlock
-	l.Ctx = ctx
-	l.FinalizedBlock = finalizedBlock
+// WithIncreasedAttestedSlot Specifies the number of slots to increase the attested slot by. This does not affect the finalized block's slot if there is any.
+func WithIncreasedAttestedSlot(increaseAttestedSlotBy int) LightClientOption {
+	return func(l *TestLightClient) {
+		l.increaseAttestedSlotBy = increaseAttestedSlotBy
+	}
+}
 
-	return l
+// WithIncreasedFinalizedSlot Specifies the number of slots to increase the finalized slot by. This DOES NOT affect the attested block's slot. That should be handled separately using WithIncreasedAttestedSlot.
```

Comment:
```suggestion
// WithIncreasedFinalizedSlot specifies the number of slots to increase the finalized slot by. This DOES NOT affect the attested block's slot. That should be handled separately using WithIncreasedAttestedSlot.
```
Can you maybe unify docs for this function and `WithIncreasedAttestedSlot`? The two are two parts of the same coin yet the descriptions are different

Reply:
they are inherently a bit different. if you just use `WithIncreasedAttestedSlot` it's fine. but if you use the `WithIncreasedFinalizedSlot` you need to be careful not to go over the attested slot, based on the fork epochs that you are mocking in your test. what do you think?